codice/ddf
# DDF-4756 Port over Solr Config files to enable Spellcheck out of the box

#### What does this PR do?
Adjusts solrconfig and schema and adds a dictionary file to support spellcheck out of the box inside DDF.

#### Who is reviewing it? 
@GabrielFabian 
@pklinef 
@derekwilhelm 
@bellcc 
@andrewkfiedler 

#### Select relevant component teams: 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@pklinef
@tbatie

#### How should this be tested?
Profile:install and make sure the dictionary file is placed inside the conf directory. 
Then open up solr UI and run https://localhost:8994/solr/catalog/select?q=*:*&spellcheck.dictionary=filebased&spellcheck=on&spellcheck.build=true inside your browser.
Then go to the admin console to turn on spellcheck under Catalog UI Search
Then go to Intrigue, ingest data and perform searches on Title/Description to see if spellcheck works as expected

#### Any background context you want to provide?
https://github.com/codice/ddf/pull/4698

#### What are the relevant tickets?
For GH Issues:
Fixes: #4756 

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
